src/transformers/models/deta/modeling_deta.py

```diff
@@ -1771,6 +1779,7 @@ class DetaForObjectDetection(DetaPreTrainedModel):
     _tied_weights_keys = [r"bbox_embed\.\d+"]
     # We can't initialize the model on meta device as some weights are modified during the initialization
     _no_split_modules = None
+    supports_gradient_checkpointing = True
```

Comment:
This should be set under the `DetaPretrainedModel` class. Once that's set there, then you can remove the custom implementations of `test_gradient_checkpointing`, as it will run the tests in `test_modeling_common` [here](https://github.com/huggingface/transformers/blob/f40b87de0ca234df61f76928956c4a2118c0b548/tests/test_modeling_common.py#L351)

Reply:
@amyeroberts I didn't realise that! Thanks for letting me know this, it was so simple I also removed custom implementation also!